File: regions/system/library/csharp/telemetry/TelemetryConfig.cs
namespace K1s0.System.Telemetry;

public sealed record TelemetryConfig
{
    public string ServiceName { get; init; } = string.Empty;

    public string Version { get; init; } = string.Empty;

    public string Tier { get; init; } = string.Empty;

    public string Environment { get; init; } = string.Empty;

    public string? Endpoint { get; init; }

    public double SampleRate { get; init; } = 1.0;

    public string LogLevel { get; init; } = "Information";

    public string LogFormat { get; init; } = "json";
}

File: regions/system/library/swift/telemetry/Sources/K1s0Telemetry/TelemetryConfig.swift
/// テレメトリ設定。
public struct TelemetryConfig: Sendable {
    public let serviceName: String
    public let version: String
    public let tier: String
    public let environment: String
    public let traceEndpoint: String?
    public let sampleRate: Double
    public let logLevel: LogLevel

    public init(
        serviceName: String,
        version: String,
        tier: String,
        environment: String,
        traceEndpoint: String? = nil,
        sampleRate: Double = 1.0,
        logLevel: LogLevel = .info
    ) {
        self.serviceName = serviceName
        self.version = version
        self.tier = tier
        self.environment = environment
        self.traceEndpoint = traceEndpoint
        self.sampleRate = sampleRate
        self.logLevel = logLevel
    }
}

/// ログレベル。
public enum LogLevel: String, Sendable {
    case debug
    case info
    case warn
    case error
}
